src/Orchard.Web/Modules/Orchard.Tokens/Tests/ListTokenTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System.Collections.Generic;
+using Autofac;
+using NUnit.Framework;
+using Orchard.ContentManagement;
+using Orchard.Tokens.Implementation;
+using Orchard.Tokens.Providers;
+	 
+namespace Orchard.Tokens.Tests {
+	[TestFixture]
+	public class ListTokenTests {
+	    private IContainer _container;
+	    private ITokenizer _tokenizer;
+	 
+	    [SetUp]
+	    public void Init() {
+	        var builder = new ContainerBuilder();
+	        builder.RegisterType<Tokenizer>().As<ITokenizer>();
+	        builder.RegisterType<TokenManager>().As<ITokenManager>();
+	        builder.RegisterType<ListTokens>().As<ITokenProvider>();
+	        builder.RegisterType<TestTokenProvider>().As<ITokenProvider>();
+	        //builder.RegisterType<Work<Tokenizer>>().As<Work<ITokenizer>>();
+	        _container = builder.Build();
+	        _tokenizer = _container.Resolve<ITokenizer>();
+	    }
+	 
+	    [Test]
+	    public void TestListForEachTokens()
+	    {
+            Assert.That(_tokenizer.Replace("{List.ForEach:{Content.Id} }", new { List = new List<IContent> { new TestUser { Id = 5 }, new TestUser { Id = 10 } } }), Is.EqualTo("5 10 "));
```

Comment:
This looks more like a `Join` than a foreach, and then it could use a separator parameter. Empty being the default.